### PR TITLE
93 rename interfaces

### LINKED ISSUE #93: Rename all interfaces consistently
Open for voting: 
:+1: for naming all interfaces with an `I` prefix (like `ISemanticEvent` or `IDatabaseAdaptable`)
:-1: for naming all interface without any prefix (like `SemanticEvent` or `DatabaseAdaptable`)
Any other suggestions? Leave them in the comments :)

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #93 

I removed the Hungarian notation in the interfaces but in order to do that I had to make a small change:
- Instead of 6 implementing classes, I have created 1 Factory that can instantiate all types of events
- Instead of creating a new EventObject with e.g. `new SemanticEvent(...)`, you now call `EventFactory.semantic(...)`.

Is not hard to review, so have fun :D

### CODE REVIEW

File: src/octopeer-github/main/Database/EventObject/EventFactory.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/**
+ * Created by Maarten on 31-05-2016.
+ *
+ * The utility class EventFactory can create the different EventObjects.
+ * It seems redundant to have this, since TypeScript can infer an interface on any anonymous object.
+ * However, so is stating the fieldnames of the objects every time a new EventObject is created.
+ * These factory methods allow EventObjects to be instantiated based on an ordered tuple (parameter list) instead.
+ */
+abstract class EventFactory {
```
Comment (MathiasMeuleman) [inline src/octopeer-github/main/Database/EventObject/EventFactory.ts]: I think some TSDoc is appropriate for the functions of this factory

**Comment by MathiasMeuleman** [comment]
Kill all the Hungarian notiation!!
Looks good :)
Thanks for adding the TSDoc, this seems ready to be merged

File: src/octopeer-github/main/Database/EventObject/MousePositionEvent.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Created by Maarten on 26-05-2016.
+ */
+
+/**
+ * An MousePositionEvent contains the data that should be posted to a Database.
+ */
+interface MousePositionEvent {
+    /** position_x and position_y specify the position on the page (i.e. relative to the top-left corner of the document). */
```
Comment (thervh70) [inline src/octopeer-github/main/Database/EventObject/MousePositionEvent.ts]: I'm not really a fan of inline comments. Especially not in such small interfaces. Consider moving the comments to the method TSDoc (in all EventObject interfaces)

**Comment by thervh70** [comment]
Nice solution, yet I have added a comment :P

**Comment by thervh70** [comment]
Looking good now, 2 aprovals so I will merge!

**Pull request merged**